tests/unit/customizations/test_overridesslcommonname.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from awscli.testutils import mock, unittest
+from awscli.testutils import BaseAWSCommandParamsTest
+from awscli.customizations.overridesslcommonname import update_endpoint_url
+
+
+class TestOverrideSslCommonName(BaseAWSCommandParamsTest):
```

Comment:
Since it is a known finite set of test cases, we should make a parametrized functional tests that goes through every case using: https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
sure I'll update